File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/gui/components/DirtBackground.java
package pize.tests.minecraft.game.gui.components;

import pize.Pize;
import pize.graphics.gl.Wrap;
import pize.graphics.texture.Region;
import pize.graphics.texture.Texture;
import pize.graphics.util.batch.TextureBatch;
import pize.gui.constraint.Constraint;
import pize.tests.minecraft.game.Session;

public class DirtBackground extends MComponent{
    
    private final Texture texture;
    
    public DirtBackground(Session session){
        this.texture = session.getResourceManager().getTexture("options_background").texture;
        this.texture.parameters.setWrap(Wrap.REPEAT);
        this.texture.update();
        
        this.setSize(Constraint.pixel(1));
    }
    
    @Override
    protected void render(TextureBatch batch, float x, float y, float width, float height){
        final int size = 5;
        
        batch.setColor(0.25, 0.25, 0.25, 1);
        batch.draw(texture, 0, 0, Pize.getWidth(), Pize.getHeight(), new Region(0, 0,
            Pize.getWidth() / (8F * size * width),
            Pize.getHeight() / (8F * size * height)
        ));
        batch.setColor(1, 1, 1, 1);
    }
    
}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/gui/components/DirtBackground.kt
package pize.tests.minecraft.game.gui.components

import pize.Pize
import pize.graphics.gl.Wrap
import pize.graphics.texture.Region
import pize.graphics.texture.Texture
import pize.graphics.util.batch.TextureBatch
import pize.gui.constraint.Constraint.Companion.pixel
import pize.tests.minecraft.game.Session

class DirtBackground(session: Session) : MComponent() {
    private val texture: Texture?

    init {
        texture = session.resourceManager.getTexture("options_background").texture
        texture.parameters.setWrap(Wrap.REPEAT)
        texture.update()
        this.setSize(pixel(1.0))
    }

    override fun render(batch: TextureBatch, x: Float, y: Float, width: Float, height: Float) {
        val size = 5
        batch.setColor(0.25, 0.25, 0.25, 1.0)
        batch.draw(
            texture, 0f, 0f, Pize.width.toFloat(), Pize.height.toFloat(), Region(
                0f, 0f,
                Pize.width / (8f * size * width),
                Pize.height / (8f * size * height)
            )
        )
        batch.setColor(1.0, 1.0, 1.0, 1.0)
    }
}